benchmarks/nox_asv_plugin.py

```diff
@@ -0,0 +1,257 @@
+"""
+ASV plug-in providing an alternative ``Environment`` subclass, which uses Nox
+for environment management.
+
+"""
+from importlib.util import find_spec
+from pathlib import Path
+from shutil import copy2, copytree
+from tempfile import TemporaryDirectory
+
+from nox.sessions import _normalize_path
+import yaml
+
+from asv.config import Config
+from asv.console import log
+from asv.plugins.conda import Conda, _find_conda
+from asv.repo import get_repo, Repo
+from asv import util
```

Comment:
I feel like `util` is a bit generic as a module name to be directly importing, (especially given you are also importing from `importlib.util`). It may clarify the code slightly to follow the pattern of `from importlib.util import find_spec` and have something more like `from asv.util import check_output, UserError`.

Reply:
I can see what you're saying, but I also think `check_output` and `UserError` are confusingly generic, so I've gone for a different compromise:

https://github.com/SciTools-incubator/iris-esmf-regrid/blob/cedcaa4438a1786c02bf41a63e58445af369c4b5/benchmarks/nox_asv_plugin.py#L18

(cedcaa4438a1786c02bf41a63e58445af369c4b5)